include/mqt-core/dd/Package.hpp

```diff
@@ -1165,6 +1165,71 @@ template <class Config> class Package {
     return '1';
   }
 
+  mEdge buildMeasOp(const Qubit index, const char measureFor = '0') {
+    mEdge measOp = {};
+    mEdge f = {};
+
+    const auto identityMatrix =
+        std::array{mEdge::one(), mEdge::zero(), mEdge::zero(), mEdge::one()};
+
+    const auto measureMatrix = measureFor == '0'
+                                   ? std::array{mEdge::one(), mEdge::zero(),
+                                                mEdge::zero(), mEdge::zero()}
+                                   : std::array{mEdge::zero(), mEdge::zero(),
+                                                mEdge::zero(), mEdge::one()};
+
+    if (index == 0) {
+      measOp = makeDDNode(static_cast<dd::Qubit>(0), measureMatrix);
+    } else {
+      measOp = makeDDNode(static_cast<dd::Qubit>(0), identityMatrix);
+    }
+
+    for (dd::Qubit p = 1; p < static_cast<dd::Qubit>(qubits()); p++) {
+      if (p == index) {
+        f = makeDDNode(static_cast<dd::Qubit>(0), measureMatrix);
+      } else {
+        f = makeDDNode(static_cast<dd::Qubit>(0), identityMatrix);
+      }
+      measOp = kronecker(f, measOp);
+    }
+    return measOp;
+  }
```

Comment:
This method should not be needed. You can just use a code sequence similar to https://github.com/cda-tum/mqt-core/blob/fc7d8660e6d2007befbcdf70abdad0e845e67ae8/include/mqt-core/dd/Package.hpp#L1181-L1184
to implement this in a way more compact fashion.

Reply:
I changed the code as suggested.

Reply:
That will not yield the correct result in general. You might have instantiated a package with 5 qubits, but only use 3 qubits in a simulation.
You'd have to use the variable index of the DD to get the proper qubit number.

In the same spirit, you can just completely inline this method as you are only using it in a single place.

Reply:
Good catch. I fixed the code as you suggested.